python3Packages.ihatemoney: 4.1 -> 4.2

symphorien opened the pull request:
Update.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

symphorien committed:
python3Packages.ihatemoney: 4.1 -> 4.1.3

symphorien committed:
nixos/ihatemoney: work around bug in uwsgi

happens when dowbloading csv reports

symphorien committed:
nixos/tests/ihatemoney: fix

privateseabass commented:
What is the package name? I'm not finding it as "ihatemoney".

symphorien commented:
I fixed the title of the PR. The commit messages are correct.

symphorien committed:
python3Packages.ihatemoney: 4.1.3 -> 4.2

symphorien committed:
python3Packages.sqlalchemy-i18n: init at 1.0.3

symphorien committed:
python3Packages.debts: init at 0.5

symphorien committed:
python3Packages.sqlalchemy-continuum: init a 1.3.9

symphorien commented:
I updated ihatemoney to 4.2 but haven't tested enough yet.

symphorien committed:
python3Packages.ihatemoney: disable on python2

it does not build and is not supported according to setup.cfg

symphorien committed:
python3Packages.ihatemoney: fix postgresql test

symphorien commented:
@GrahamcOfBorg build python3.pkgs.ihatemoney.tests

symphorien commented:
@GrahamcOfBorg build python3.pkgs.ihatemoney.tests.ihatemoney.ihatemoney-postgresql  python3.pkgs.ihatemoney.tests.ihatemoney.ihatemoney-sqlite

symphorien commented:
Looks like the PR is ready.

symphorien commented:
/marvin opt-in

symphorien commented:
/status needs_review

timokau commented on pkgs/development/python-modules/debts/default.nix:
```diff
@@ -0,0 +1,32 @@
+    domain = "framagit.org";
+    owner = "almet";
+    repo = "debts";
+    rev = "09ee167c"; # no tags
```
That's a shame. In that case we should at the very least use a fully-qualified rev instead of a shortened one.

But I kind of think "no tags (-> no automatic updates)" outweighs "no tests", so I'd rather fetch it from pypi for now. Maybe create an issue against the upstream repo, notifying them that tags would be useful :)

symphorien replied:
changed to full revision.

timokau commented on pkgs/development/python-modules/debts/default.nix:
```diff
@@ -0,0 +1,32 @@
+  propagatedBuildInputs = [ jinja2 ];
+
+  # for some reason tests only work if the module is properly installed
+  doCheck = false;
```
Even more reason to just fetch it from pypi.

timokau commented on pkgs/development/python-modules/debts/default.nix:
```diff
@@ -0,0 +1,32 @@
+  postInstall = ''
+    rm -r debts
+    export PYTHONPATH=$out/${python.sitePackages}:$PYTHONPATH
+    python nix_run_setup test
```
Why `postInstall`? Seems like this should be `checkPhase`. Note that the python `checkPhase` is actually `installCheckPhase`, so its always run after installation.

symphorien replied:
Oh I didn't know checkPhase was after installPhase for python packages. I reworked this.

timokau commented on pkgs/development/python-modules/sqlalchemy-i18n/default.nix:
```diff
@@ -0,0 +1,29 @@
+{ stdenv, lib, fetchPypi, buildPythonPackage
```
The mix in styles (one per line vs long lines) irks me a bit here :smile:

jonringer replied:
I think I started doing this, everything but python packages are included in the first line, python packages are then entered 1 per line.

Generally `stdenv, lib, fetchPypi, buildPythonPackage` are unlikely to ever change, so just have them together.

I guess it's just a personal preference

symphorien replied:
I ran nixpkgs-fmt on the file :)

timokau commented on nixos/modules/services/web-apps/ihatemoney/default.nix:
```diff
@@ -132,6 +132,9 @@ in
             env = [ "IHATEMONEY_SETTINGS_FILE_PATH=${configFile}" ];
             pythonPackages = self: [ self.ihatemoney ];
+            # workaround for https://github.com/unbit/uwsgi/issues/1126
+            # probably fixed in uwsgi 2.0.19 https://github.com/unbit/uwsgi/commit/d0444450a195f218456a2149d9c2066afa0ae52a
```
Released 2 days ago! https://github.com/unbit/uwsgi/releases

Probably better to make that update first & remove this workaround, since otherwise it will likely stick around forever.

symphorien replied:
Do you want me to update uwsgi in this PR ?

timokau replied:
Either that, or in its own PR (and ping me for review), or hope that somebody else does it soon ;)

timokau reviewed (changes requested):
Looks mostly good, I left some comments.
/status needs_work

timokau commented:
/status needs_work
That should work in review comments too.

timokau commented:
Also the testsuite fails for me in `nix-review`.

symphorien commented:
Ah I can reproduce the test failure when I rebase on master.
A git bisect indicates that it's caused by 86990d60071a3be8ea25d2ad2f79a0ba096331b6     pythonPackages.wtforms: 2.1 -> 2.3.1
and apparently ihatemoney only supports 2.2.1: https://github.com/spiral-project/ihatemoney/commit/53dae96bdbfcf09f4cef56afda8a7ad567458eeb

I wonder what is the way to go from there.

symphorien commented:
upstream knows about the problem https://github.com/spiral-project/ihatemoney/issues/567

timokau commented:
Pinning dependencies in python is generally messy. For a python application we *could* get away with it, for a library we can't.

Probably should have caught this in the initial review: Is this really a library? Looks more like an application to me. Search https://nixos.org/nixpkgs/manual/ for "Building packages and applications" for more info.

Since upstream knows about it we could also
- wait for a fix there and apply the patch
- fix it for them, create a PR and *then* apply the patch.

Depending on how much work you want to invest.

symphorien commented:
> Probably should have caught this in the initial review: Is this really a library? 

It is a wsgi application. That is, uwsgi starts a special python interpreter, and then *imports* ihatemoney. So for all intents and purposes, it is a library.

Regarding patching, I had a quick look, and have no idea what happens.
And regarding waiting, current version of ihatemoney is also affected.

I'll try and see if it still works if I disable this test in the coming days.

Mic92 commented:
Result of `nixpkgs-review pr 85023` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
<br>- python37Packages.ihatemoney
<br>- python38Packages.ihatemoney
</details>
<details>
  <summary>8 packages built:</summary>
<br>- python27Packages.sqlalchemy-continuum
<br>- python27Packages.sqlalchemy-i18n
<br>- python37Packages.debts
<br>- python37Packages.sqlalchemy-continuum
<br>- python37Packages.sqlalchemy-i18n
<br>- python38Packages.debts
<br>- python38Packages.sqlalchemy-continuum
<br>- python38Packages.sqlalchemy-i18n
</details>

Mic92 commented:
@GrahamcOfBorg build python37Packages.ihatemoney python38Packages.ihatemoney

timokau commented:
Mic92: There is a known problem with a too-new dependency.

FRidh commented on pkgs/development/python-modules/debts/default.nix:
```diff
@@ -0,0 +1,42 @@
+
+  checkInputs = [ pytest ];
+
+  # for some reason tests only work if the module is properly installed
```
Yes this is actually a problem in `buildPythonPackage`. We should remove the folder always or change directory.

symphorien committed:
remove workaround for uwsgi < 2.0.19

symphorien committed:
nixos/tests/ihatemoney: fix

symphorien committed:
uwsgi: 2.0.18 -> 2.0.19.1

release notes:
https://uwsgi-docs.readthedocs.io/en/latest/Changelog-2.0.19.html
https://uwsgi-docs.readthedocs.io/en/latest/Changelog-2.0.19.1.html

symphorien committed:
python3Packages.ihatemoney: pin wtforms to 2.2.1

symphorien commented:
I thought a bit about this, and came to the conclusion that I can override the dependencies of ihatemoney locally because ihatemoney will never be the dependency of something else. It is only a library because it must be imported by uswgi, but it will not be depended upon by some other python module.

I also updated uswgi and removed the now-unneeded workaround. I tested this in a vm, and csv downloading works :)

/status needs_review

timokau commented:
I agree with your reasoning, since this is a de-facto python application. Still, since we kind of have an (unwritten?) policy against dependency overrides within python-module I'd like @FRidh's okay on this.

@FRidh do you agree that its fine to override the dependencies here, since the module will only ever be used as an application (but through uswgi)?

timokau commented:
@jonringer do you maybe have an opinion on this? See my last comment.

/status needs_work
(work = feedback to get it out of the needs_review queue, will be improved by https://github.com/timokau/marvin-mk2/issues/29)

timokau commented:
Well, lacking any feedback I will just assume its fine. Please add the `python.pkgs` prefix to https://github.com/NixOS/nixpkgs/pull/85023/commits/ebc28de6a9001052abdc6bd6141ce0c228275d41, then this should be good to go.

I would still prefer to avoid hardcoding the `debts` rev to enable automatic updates, but If you as the maintainer want to stay on top of updates manually, that is your call.

symphorien commented:
done. The nixos tests still pass locally.

timokau commented:
Thanks, and thank you for your patience!
